### PR TITLE
[Resolution] Clean up orphans when swapping

### PR DESCRIPTION
**Pull request by segiddins**
Closes https://github.com/bundler/bundler/issues/4198

- [x] Docs
- [x] CHANGELOG

\c @indirect

### CODE REVIEW

File: lib/molinillo/resolution.rb [hunk 1]
```diff
@@ -341,26 +341,30 @@ def attempt_to_activate_existing_spec(existing_node)
...
+        actual_vertex.payload = possibility
+        fixup_swapped_children(actual_vertex)
+        activate_spec
+      end
+
+      def fixup_swapped_children(vertex)
+        payload = vertex.payload
+        dep_names = dependencies_for(payload).map(&method(:name_for))
+        vertex.successors.each do |succ|
+          activated.detach_vertex_named(succ.name) if !dep_names.include?(succ.name) && !succ.root? && succ.predecessors.to_a == [vertex]
```
Comment (segiddins) [inline lib/molinillo/resolution.rb]: debug here

**Comment by segiddins** [comment]
@indirect care to review this?

**Comment by indirect** [comment]
I really only have a faint and foggy notion of what this is doing, but the code looks quite nice to me. :stuck_out_tongue_closed_eyes:

**Comment by segiddins** [comment]
This should be good to merge!

**Pull request merged**